common/src/main/java/com/wynntils/services/mapdata/MapFeatureService.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.mapdata;
+
+import com.google.gson.reflect.TypeToken;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Service;
+import com.wynntils.core.components.Services;
+import com.wynntils.core.net.Download;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.services.mapdata.features.CombatLocation;
+import com.wynntils.services.mapdata.features.PlaceLocation;
+import com.wynntils.services.mapdata.features.ServiceLocation;
+import java.util.List;
+
+public class MapFeatureService extends Service {
+    public MapFeatureService(MapDataService mapData) {
+        super(List.of(mapData));
+
+        loadPlaces();
+        loadServices();
+        loadCombatLocations();
+    }
+
+    private void loadPlaces() {
+        Download dl = Managers.Net.download(UrlId.DATA_STATIC_MAPDATA_PLACES);
+        dl.handleReader(reader -> {
+            TypeToken<List<PlaceLocation>> type = new TypeToken<>() {};
+            List<PlaceLocation> places = Managers.Json.GSON.fromJson(reader, type.getType());
+            Services.MapData.PLACE_LIST_PROVIDER.updatePlaces(places);
```

Comment:
I had some kind of idea that the Providers themselves should be responsible for loading the json file. The `update` hooks were just a way to get them to work with the existing framework.

Maybe it does not turn out good, but that was at least my idea.

Reply:
So, like, either in the constructor of e.g. CombatListProvider, or in MapDataService, when we do `registerBuiltInProvider(COMBAT_LIST_PROVIDER)`.

Reply:
I've been thinking about something like this, but I've failed to turn out anything solid. Also, I just remember that I did not implement the "reload" hook in this service.

Reply:
But you intend to look at this? Or did you mean that you thought that my suggestion was a bad idea that was too hard to implement?

Reply:
> But you intend to look at this? Or did you mean that you thought that my suggestion was a bad idea that was too hard to implement?

I had the idea to use something like `createOnlineProvider` to download these. But I really have no idea how they are meant to be used, and I have yet to figure it out, because it's only strictly needed for porting "poi providers". All in all, I think with `MapFeatureService` does not look too bad, so if we keep it as is, I won't have sleepless nights about it.

Reply:
Ok, let's take this in as it is, and then I can play around a bit with it and see if I find some different solution I'm more happy with.